### DISCUSSION TITLE
`consult-yank-from-kill-ring` appends the text, instead of inserting it

### DISCUSSION THREAD

**Discussion by sergiodj** [minad/consult#1029] [Q&A] [answered]
Hi,

I'm using version `20240510.1302`.

Maybe I'm using the command wrongly, but when I invoke `consult-yank-from-kill-ring` and select the text I want to yank, it inserts the text at the front of the first yanked text instead of only inserting the selected text.  For example, if I press `C-y` and the first text in the kill ring is `first text`, I correctly see:

```
first text
```

Then, if I press `M-y` and select another text from the kill ring (for example, `third text`), I see:

```
first textthird text
```

My configuration is pretty much what's suggested from the README file, but I can provide it if you can't reproduce the problem.

Thanks.

**Comment by minad** [comment]
Can you please describe the exact steps you perform? I have difficulties understanding the problem.

**Reply by sergiodj** [reply]
Hi,

Hm, OK, I will try to expand on the steps described above.

1) Write the following on a scratch buffer:

```
first text
second text
third text
```

2) Kill each line with `C-k` from the bottom to the top.  This will make `first text` be the first entry in the kill ring, `second text` be the second entry, and `third text` be the third entry.

3) Now, yank the first entry from the kill ring with `C-y`.  You will see `first text` be inserted.

4) Invoke `consult-yank-from-kill-ring` with `M-y`.  Select `third text` and hit `RET`.

5) I expected to see `first text` be replaced with `third text` in the buffer.  Instead, I see `first textthird text`.

**Reply by sergiodj** [reply, marked as answer]
It seems like I've made some confusion about the commands here.  `consult-yank-from-kill-ring` just does a yank; what I'm really looking for here seems to be `consult-yank-replace` or `consult-yank-pop`.
